src/Features/Core/Portable/AddParameter/AbstractAddParameterCodeFixProvider.cs

```diff
@@ -111,74 +154,164 @@ internal abstract class AbstractAddParameterCodeFixProvider<
         {
-            var methodDeclaration = await method.DeclaringSyntaxReferences[0].GetSyntaxAsync(cancellationToken).ConfigureAwait(false);
+            var newSolution = invocationDocument.Project.Solution;
+            
+            //method.DeclaringSyntaxReferences does no longer return the two locations for partial methods so this is no longer true:
+            //https://stackoverflow.com/a/38294041
+            //Therefore we need this ugly hack:
+            var methodSymbolReferences =
+                Enumerable.Repeat(method, 1)
+                .Union(Enumerable.Repeat(method.PartialDefinitionPart, 1)
+                .Union(Enumerable.Repeat(method.PartialImplementationPart, 1)))
+                .Where(methodSymbol => methodSymbol != null)
+                .Distinct().ToImmutableArray();
+
+            foreach (var methodSymbol in methodSymbolReferences)
+            {
+                var methodDeclaration = await methodSymbol.DeclaringSyntaxReferences[0].GetSyntaxAsync(cancellationToken).ConfigureAwait(false);
+                var methodDeclarationRoot = methodDeclaration.SyntaxTree.GetRoot(cancellationToken);
+                var methodDocument = invocationDocument.Project.Solution.GetDocument(methodDeclaration.SyntaxTree);
+                var syntaxFacts = methodDocument.GetLanguageService<ISyntaxFactsService>();
+                var editor = new SyntaxEditor(methodDeclarationRoot, methodDocument.Project.Solution.Workspace);
+                var parameterDeclaration = default(SyntaxNode);
+                var insertionIndex = default(int);
+                var newTypeParameter = newTypeArgument == null ? null : CreateTypeParameterSymbol(method);
+
+                if (argument != null)
+                {
+                    (parameterDeclaration, insertionIndex) = await CreateParameterDeclarationAsync(editor, invocationDocument, method, methodDeclaration, argumentList, argument, newTypeArgument, newTypeParameter, cancellationToken).ConfigureAwait(false);
+                }
 
-            var (parameterSymbol, isNamedArgument) = await CreateParameterSymbolAsync(
-                invocationDocument, method, argument, cancellationToken).ConfigureAwait(false);
+                AddParameter(
+                    syntaxFacts, editor, method, methodDeclaration, argument,
+                    insertionIndex, parameterDeclaration, newTypeParameter, cancellationToken);
 
-            var methodDocument = invocationDocument.Project.Solution.GetDocument(methodDeclaration.SyntaxTree);
-            var syntaxFacts = methodDocument.GetLanguageService<ISyntaxFactsService>();
-            var methodDeclarationRoot = methodDeclaration.SyntaxTree.GetRoot(cancellationToken);
-            var editor = new SyntaxEditor(methodDeclarationRoot, methodDocument.Project.Solution.Workspace);
+                var newRoot = editor.GetChangedRoot();
+                newSolution = newSolution.WithDocumentSyntaxRoot(methodDocument.Id, newRoot);
```

Comment:
this loop feels like it will be wrong if both parts of the partial method are in the same document.

Reply:
Note: we will need to update all related overrides/implementations.

For examples of how we do this elsewhere, take a look at ReplaceMethodWithPropertyCodeRefactoringProvider.  Specifically ReplaceGetMethodsAndRemoveSetMethodsAsync.

Here's the basic idea:

1. You're going to do a SymbolFinder.FindReferencesAsync call to find all the cascaded methods from the method you want to update.
2. You'll ignore the actual references, and just use all the definitions that were returned.
3. You'll then group all the definitions by their defining-document.  This way you can process one doc at a time, and update all the methods in that document at once.
4. You then process each document one at a time, creating the new document.  
5. You create a new solution snapshot out of all these updated documents.

To update a document we create a single SyntaxEditor for that document, then go and process all the IMethodSymbols we grouped for that doucment.  For each IMethodSymbol we go do it's declaration and update it according, making the change to the editor.  

--

Note: if you do this, you don't need any special partial-method handling.  It will just fall out naturally from doing the FindReferences.